Bio/EBI/ebisearch.py

```diff
@@ -0,0 +1,798 @@
+
+The following code was developed by Berenice Batut (berenice.batut@gmail.com)
+
+References:
+Park Y.M., Squizzato S., Buso N., Gur T., Lopez R. (2017)
+The EBI search engine: EBI search as a service making biological data accessible
+for all.;Nucleic Acids Research, May 2, 2017; DOI: 10.1093/nar/gkx359
+
+"""
+
+from Bio._py3k import urlopen as _urlopen
+from Bio._py3k import Request
+import json
+
+
+base_url = 'http://www.ebi.ac.uk/ebisearch/ws/rest'
+size_limit = 100
+start_limit = 250000
+
+
+def get_domain_details(domain):
+    """Return a dictionary with the details of a given domain in EBI.
+
+    Arguments:
+     - domain - domain id in EBI.
+
+    Example:
+
+    >>> dom_dict = ebisearch.get_domain_details(domain="allebi")
+    >>> print(str(dom_dict['domains'][0])[:150])
+    >>> {u'description': u'Results from all the EBI resources', u'subdomains':
+        [{u'description': u'', u'subdomains': [{u'fieldInfos': [{u'description'
+        : u'Luce
+    """
+    url = base_url + '/' + domain
+    url = url.replace(" ", "%20")
+    req = Request(
+        url,
+        headers={"accept": "application/json"})
+    r = _urlopen(req)
+    if (r.getcode() == 400):
+        raise Exception("Wrong user input.")
+    if (r.getcode() == 404):
+        raise Exception("Wrong URL.")
+    if (r.getcode() == 500):
+        raise Exception("Server internal error.")
+    return json.loads(r.read().decode())
+
+
+def get_details_and_subdomains(domain, level, verbose=False):
+    """Return the details (id and name) for the domain and its subdomains.
+
+    Arguments:
+     - domain - domain id in EBI.
+     - verbose - Boolean to define the printing info.
+    """
+    domain_details = {domain['id']: domain['name']}
+    if verbose:
+        print("\t" * level + "%s -- %s" % (domain['id'], domain['name']))
+
+    if "subdomains" not in domain:
+        return domain_details
+    else:
+        for subdomain in domain["subdomains"]:
+            domain_details.update(
+                get_details_and_subdomains(subdomain, level + 1, verbose))
+    return domain_details
+
+
+def get_domains(verbose=False):
+    """Return the list of domains in EBI as a dictionary.
+
+    The key being the domain id and the value the domain name.
+    Arguments:
+
+     - verbose - boolean to define the printing info.
+    """
+    allebi = get_domain_details("allebi")
+    domain_details = {}
+    for domain in allebi["domains"]:
+        domain_details.update(get_details_and_subdomains(domain, 0, verbose))
+    return domain_details
+
+
+def print_domain_hierarchy():
+    """Print the hierarchy of the domains."""
+    get_domains(verbose=True)
+
+
+def check_domain(domain):
+    """Check if a domain exist in EBI.
+
+    Arguments:
+
+     - domain - id of a domain to check.
+    """
+    if domain not in get_domains(verbose=False):
+        err_str = "The domain does not correspond to the id of a known domain"
```

Comment:
You could write this as a multi-line string. If you did that there is no need to create the temporary variable ``err_str``.